src/main/java/matveyeva/phonebook/crud/UserCRUD.java

```diff
@@ -0,0 +1,151 @@
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public User createUser(String str){
+        try{
+            User user = split(str);
+            users.add(user);
+            logger.info("New user " + user.getUserName() + " created");
+            return user;
+        }catch (InvalidUserException ex){
+            System.out.println(ex.getMessage());
+            return null;
+        }catch (IllegalArgumentException iex){
+            System.out.println("User exists");
+            return null;
+        }
+
+//        if(user != null && user.isUserValid()){
+//            try {
+//                if(users.add(user)){
+//                    logger.info("New user " + user.getUserName() + " created");
+//
+//                    return user;
+//                }
+//            } catch (Exception e) {
+//                e.printStackTrace();
+//            }
+//        }
+//        return null;
+    }
+
+    public void delete(User user) throws IOException {
+        contactCRUD = new ContactCRUD(user);
+        contactCRUD.deleteAll();
+        try {
+            contactCRUD.reloadContacts();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        users.remove(user);
+        reloadUsers();
+    }
+
+    public void deleteAll() throws IOException {
+        for(User user: users){
+            contactCRUD = new ContactCRUD(user);
+            contactCRUD.deleteAll();
+        }
+        contactCRUD.reloadContacts();
+        ArrayList<User> arr = new ArrayList<User>(users);
+        users.removeAll(arr);
+        reloadUsers();
+    }
+
+    public User update(String newUser, User oldUser){
+        try{
+            User user = split(newUser);
+            users.remove(oldUser);
+            users.add(user);
+            return user;
+        }catch(InvalidUserException ex){
+            System.out.println(ex.getMessage());
+            return null;
+        }
+    }
+
+    public Set<User> findAll(){
+        return users;
+    }
+
+    public User findByName(String username){
+        for(User user: users){
+            if(user.getUserName().contains(username)){
+                return user;
+            }
+
+        }
+        return null;
+    }
+
+    private void loadUsers() throws Exception {
+        FileInputStream fileInputStream = new FileInputStream("users.ser");
+        while (fileInputStream.available() > 0) {
+            ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream);
+            User user = (User) objectInputStream.readObject();
+            users.add(user);
+        }
+    }
+    public void reloadUsers() throws IOException {
+        FileOutputStream outputStream = new FileOutputStream("users.ser");
+        for(User user : users){
+            ObjectOutputStream objectOutputStream = new ObjectOutputStream(outputStream);
+            objectOutputStream.writeObject(user);
+        }
+        outputStream.close();
```

Comment:
To avoid closing resources manually, you can use try-with-resources block: `try (FileOutputStream outputStream = new FileOutputStream("users.ser"))`